presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -502,9 +503,9 @@ public void testSameScalarSubqueryIsAppliedOnlyOnce()
         // same query used for left, right and complex join condition
         assertEquals(
                 countOfMatchingNodes(
-                        plan("SELECT * FROM orders o1 JOIN orders o2 ON o1.orderkey = (SELECT 1) AND o2.orderkey = (SELECT 1) AND o1.orderkey + o2.orderkey = (SELECT 1)"),
+                        plan("SELECT * FROM orders o1 JOIN orders o2 ON o1.orderkey = (SELECT 1) AND o2.orderkey = (SELECT 1) AND o1.orderkey + o2.orderkey = (SELECT 2)"),
```

Comment:
What is this change for?

Reply:
original query has 1+1=1 which is false and triggers empty join optimizations which makes one node and changes the result. Made a change to avoid the optimization with 1+1 = 2 which makes number of nodes to 2 rather than 1.